agent/dns.go

```diff
@@ -782,6 +782,11 @@ func (d *DNSServer) doDispatch(network string, remoteAddr net.Addr, req, resp *d
 
 	case "addr":
 		// <address>.addr.<suffixes>.<domain> - addr must be the second label, datacenter is optional
+
+		//check if the query type is A for IPv4 or AAAA for IPv6 or ANY
+		if (req.Question[0].Qtype != dns.TypeANY) && (req.Question[0].Qtype != dns.TypeA || len(queryParts[0])/2 != 4) && (req.Question[0].Qtype != dns.TypeAAAA || len(queryParts[0])/2 != 16) {
```

Comment:
A switch/case on `Qtype` might make this a bit easier to read.

Is the `len(queryParts[0]/2)` part necessary, or would that be handled later on by `hex.DecodeString` ? We might want to the `Qtype` check separate from the length check so that we can return different error codes.

If we need to keep this all together it would be good to extract this into a function to avoid the really long line.

Reply:
I agree, it's not readable. I will check. The reason I added the len check is for the case we have an IPv6 request with an IPv4 len.
Not sure if that's possible though.